### PR TITLE
[WebAssembly] Change half-precision feature name to fp16.

### PR DESCRIPTION
**Pull request by brendandahl**
This better aligns with how the feature is being referred to and what runtimes (V8) are calling it.

### CODE REVIEW

**Review by aheejin — COMMENTED** [review]
I know it is a little pedantic, but I recently sorted the features alphabetically (#90777)... Can you move the new `fp16` entries so that the lists are alphabetically sorted in the files?

EDIT: nevermind sorry

**Comment by dschuff** [comment]
I'm fine with calling this feature fp16 (I like it better than half-precision anyway) but I also wanted to point out that consistency with the spec is probably more important in the long term than consistency with V8, because the V8 flags eventually disappear (once V8 enabled support for a feature by default), whereas the flags for the tools will probably persist for the long term. Of course the spec names will also go away once the feature is merged into the spec (until we have profiles, I guess).... But just something to consider. For that matter, maybe we should make the feature names a tool-convention to line up with wasm-tools. I think feature naming is already kind of a mess, maybe we can avoid making it worse.

**Comment by aheejin** [comment]
> I know it is a little pedantic, but I recently sorted the features alphabetically (#90777)... Can you move the new `fp16` entries so that the lists are alphabetically sorted in the files?

Sorry nevermind, even if we want to sort it its place is the same... 😅

**Review by tlively — APPROVED** [review]

**Review by dschuff — APPROVED** [review]

File: clang/test/Preprocessor/wasm-target-features.c [hunk 1]
```diff
@@ -178,7 +178,7 @@
 // GENERIC-NOT: #define __wasm_bulk_memory__ 1{{$}}
 // GENERIC-NOT: #define __wasm_exception_handling__ 1{{$}}
 // GENERIC-NOT: #define __wasm_extended_const__ 1{{$}}
-// GENERIC-NOT: #define __wasm_half_precision__ 1{{$}}
+// GENERIC-NOT: #define __wasm__fp16__ 1{{$}}
```
[the remark this answers is not in the document]
Comment (brendandahl) [reply clang/test/Preprocessor/wasm-target-features.c]: Did you pass the flag in via the js flags? e.g. `--js-flags="--experimental-wasm-fp16"`

**Pull request merged**